File: solutions/csharp/secret-handshake/1/SecretHandshake.cs
using System.Collections.Generic;

public static class SecretHandshake
{
    private enum Code
    {
        Wink = 1,
        DoubleBlink = 2,
        CloseEyes = 4,
        Jump = 8,
        Reverse = 16,
    }

    public static string[] Commands(int commandValue)
    {
        List<string> commands = new();
        for (int i = 1; i < 16; i *= 2)
        {
            string code = GetCodeWord(commandValue & i);
            if (!string.IsNullOrEmpty(code))
                commands.Add(code);
        }
        if ((commandValue & (int)Code.Reverse) == (int)Code.Reverse)
            commands.Reverse();

        return commands.ToArray();
    }

    private static string GetCodeWord(int num) =>
        num switch
        {
            (int)Code.Wink => "wink",
            (int)Code.DoubleBlink => "double blink",
            (int)Code.CloseEyes => "close your eyes",
            (int)Code.Jump => "jump",
            _ => string.Empty,
        };
}

File: solutions/cpp/secret-handshake/1/secret_handshake.cpp
#include "secret_handshake.h"
#include <string>
#include <vector>

namespace secret_handshake {
std::vector<std::string> commands(int code) {
  std::vector<std::string> commands{};
  if ((code & 0b0001) > 0) {
    commands.emplace_back("wink");
  }
  if ((code & 0b0010) > 0) {
    commands.emplace_back("double blink");
  }
  if ((code & 0b0100) > 0) {
    commands.emplace_back("close your eyes");
  }
  if ((code & 0b1000) > 0) {
    commands.emplace_back("jump");
  }
  if ((code & 0b0001'0000) > 0) {
    std::reverse(commands.begin(), commands.end());
  }
  return commands;
}
} // namespace secret_handshake
